### PR TITLE
update webpack-cli for examples

### PR DESCRIPTION
**Pull request by sokra**
upgrade to webpack-cli 4 for building the examples

**What kind of change does this PR introduce?**
bugfix, deps

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
nothing

## Blocked by

* bug in webpack-cli that e. g. `--output-pathinfo` is not correctly applied when the webpack.config.js exports an array of configs
* `[webpack-cli] Compilation finished` should not be there.

### CODE REVIEW

File: examples/template-common.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ exports.replaceBase = (template) => {
 	return template
 		.replace(/\r\n/g, "\n")
 		.replace(/\r/g, "\n")
+		.replace(/\[webpack-cli\].+\n/g, "")
```
Comment (sokra) [inline examples/template-common.js:36]: This should be removed in webpack-cli

**Comment by sokra** [comment]
cc @alexander-akait

File: examples/template-common.js (hunk 1 above)
Comment (snitin315) [reply examples/template-common.js:36]: Other `plugins` and `loaders` can generate `warnings` and `errors` too. Hence we added `[webpack-cli]` to have a better distinction between errors & warnings. (If I remember correctly)

**Comment by alexander-akait** [comment]
@sokra yep, without `[webpack-cli]` hard to say who log information, it is mess of lines without understanding where they come from

**Comment by alexander-akait** [comment]
> bug in webpack-cli that e. g. --output-pathinfo is not correctly applied when the webpack.config.js exports an array of configs

Can you clarify?

File: examples/template-common.js (hunk 1 above)
Comment (sokra) [reply examples/template-common.js:36]: Sorry that was a bit unclear. I'm not against the prefix. I was only complaining about the `Compilation finished` message, which is redundant to the `webpack compiled successfully` message in the stats output.

A prefix is good in general for all other messages...

**Comment by sokra** [comment]
> > bug in webpack-cli that e. g. --output-pathinfo is not correctly applied when the webpack.config.js exports an array of configs
> 
> Can you clarify?

When having something like

```js
module.exports = [
{...},
{...}
]
```

in webpack config, the command line arguments have no effect. E. G. `--output-pathinfo` should switch on pathinfo for both configs.

**Comment by snitin315** [comment]
@sokra I will fix this behavior. 

Just to confirm -

```js
module.exports = [
{mode: 'none', stats:  'normal'},
{mode: 'dev',stats: true}
];
```

passing `webpack --stats verbose` should change it to following right? -

```js
module.exports = [
{mode: 'none', stats:  'verbose'},
{mode: 'dev',stats: 'verbose'}
];
```

**Comment by sokra** [comment]
Yep

**Pull request merged**